src/IB/IBFEMethod.cpp

```diff
@@ -929,6 +998,13 @@ IBFEMethod::spreadForce(const int f_data_idx,
     batch_vec_copy({ d_X_half_vecs, d_F_half_vecs }, { d_X_IB_ghost_vecs, d_F_IB_ghost_vecs });
     batch_vec_ghost_update({ d_X_IB_ghost_vecs, d_F_IB_ghost_vecs }, INSERT_VALUES, SCATTER_FORWARD);
 
+    if (d_use_scratch_hierarchy)
+    {
+        // TODO: this assumes the structure is always on the finest level
+        // TODO: we don't need a RefinePatchStrategy here, right?
```

Comment:
You do if you want physical boundary values filled.

Reply:
Yes, but this is never relevant with computed forces, right?

Reply:
For spreading as currently implemented, the `FEDataManager::spread()` routine needs to know what the physical BCs are (to accumulate from the physical boundary ghost values back into the interior), but I don't think that needs to be used to transfer from the scratch hierarchy back to the primary hierarchy.

Reply:
I think my question is, more exactly, since there are no boundary conditions for forces we do not need to do this when moving data to the scratch hierarchy, right?

Reply:
Ah. It depends.

If for some reason the forces need to be interpolated from coarser levels to fill in the fine grid level, and if that interpolation needs to happen near the boundary, then you will need some ghost values...

However, I think that we want to try to ensure that the scratch hierarchy is contained in the corresponding fine level of the primary hierarchy, so that data transfer is just straight-up copying in the overlap regions.

Reply:
The scratch hierarchy should always occupy the same region of space as the primary hierarchy so I think this is fine.